dotcom-rendering/src/web/components/FrontMostViewed.tsx

```diff
@@ -0,0 +1,63 @@
+import type { DCRFrontCard } from '../../types/front';
+import type { TrailTabType, TrailType } from '../../types/trails';
+import { Island } from './Island';
+import { MostViewedFooter } from './MostViewedFooter.importable';
+import { MostViewedFooterLayout } from './MostViewedFooterLayout';
+
+type Props = {
+	trails: DCRFrontCard[];
+	mostViewed: TrailType[];
+	mostCommented?: TrailType;
+	mostShared?: TrailType;
+	displayName: string;
+	isNetworkFront: boolean;
+};
+
+export const FrontMostViewed = ({
+	trails,
+	mostViewed,
+	mostCommented,
+	mostShared,
+	displayName,
+	isNetworkFront,
+}: Props) => {
+	const showMostViewedTab = !isNetworkFront && !!mostViewed.length;
+	const sectionName = displayName.replace('Most viewed ', '');
+
+	const tabs: TrailTabType[] = [
+		{
+			heading: showMostViewedTab ? sectionName : undefined,
+			trails: trails.slice(0, 10),
+		},
+	];
+
+	if (showMostViewedTab) {
+		tabs.push({
+			heading: 'Across the guardian',
+			trails: mostViewed,
+		});
+	}
+
+	return (
+		<MostViewedFooterLayout>
+			{/* We only need hydration if there are multiple tabs */}
```

Comment:
Nice optimisation.

Reply:
Do we also need to check that `tabs.length > 1` ?

Reply:
I don't think so? we'll push to the tabs array if `showMostViewedTab` is true (well truthy), and showMostViewedTab will only be true if there are items in the mostViewed array, so I think it'd be impossible for there _not_ to be 2 tabs when showMostViewedTab is true?